src/li.jl

```diff
@@ -0,0 +1,205 @@
+            Θ = 1
+        else
+            Θ = 0
+        end
+        # println("G = $G, F=$F, Θ=$Θ, A=$A")
+        return ( G - F - Θ*A )
+    else 
+        # power series around mu=0, for z = e^mu
+        Li_series_mu(s, z, accuracy)
+    end
+end
+     
+
+"""
+    Polylog(s, z)
+
+    This is just an alias of Li(s, z)
+
+"""
+polylog(s::Number, z::Number ) = Li(s, z)
+
+    
+####################################################################
+#### these are component functions and aren't exported at this point
+     
+# Dirichlet beta function, for testing results
+#   https://en.wikipedia.org/wiki/Dirichlet_beta_function
+#   but don't export as it would create confusion with Euler integral     
+function Dbeta(s::Number)
+    β = 4.0^(-s) * ( zeta(s,0.25) - zeta(s,0.75) )
+end
+    
+function Li_direct(s::Number, z::Number, accuracy=1.0e-18)
+    # calculate using direct definition
+    if abs(z) > 1 || ( abs(z) ≈ 1  && real(s) <= 2)
+        warn("Should have |z| < 1 or (|z|==1 and Re(s)>2)")
+        throw(DomainError)
+    end
+    if abs(z) > 1/2
+        warn("Slow convergence for  |z| > 1/2")
+    end
+    total = 0.0
+    L = ceil(-log10(accuracy)*log2(10)) # summation limit from Crandall, which is conservative
+    for n=1:L
+        a = z^n / n^s
+        total += a
+        # println("   total = $total")
+    end
+    return total
+end
+
+function Li_series_mu(s::Number, z::Number, accuracy=1.0e-18)
+    # calculate using power series around μ = log(z) = 0
+    μ = log(complex(z))
+    # println("μ = $μ") 
+    if abs(μ) > 2*pi
+        warn("Should have |μ| < 2*pi, but |μ| = $(abs(μ))")
+        throw(DomainError)
+    end
+    L = Int(ceil(-log10(accuracy)*log2(10))) # summation limit from Crandall, which is conservative
+    if isinteger(s)
+        n = Int(round(s))
+        if n>0
+            # Crandall's 1.4 for s integer
+            total = μ^(n-1)*(harmonic(n-1) - log(-μ))/gamma(n)
+            # println("   μ=$μ, total = $total")
+            for m=0:L
+                if n - m != 1
+                    total += μ^m * zeta(n - m) / gamma(m+1)
+                end
+                # println("   m=$m, total = $total")
+            end
+            # println("   μ=$μ, total = $total")
+            A = 2*pi*im*log(complex(z))^(s-1)/gamma(n)
+            if  isreal(z) && real(z)>=1 
+                total -= A
+            end
+            # println("   μ=$μ, total = $total")
+        elseif n==0
+            total = z ./ (1-z)
+        elseif n<0
+            # Crandall's 1.5 for s integer 
+            total = factorial(-n) * (-μ)^(n-1)
+            for k=0:L
+                total -= μ^k * bernoulli(k-n+1, 0.0) / ( gamma(k+1)*(k-n+1) )
+            end
+        else
+            error("Should not get this case")
+        end
+    else
+        # equivalent of Crandalls 1.4 for s non-integer 
+        total = gamma(1-s) * (-μ)^(s-1)
+        for k=0:L
+            total += μ^k * zeta(s-k)/factorial(Float64(k))
```

Comment:
More terrible ways to compute power series.  You should never have to call `factorial` or `gamma` etcetera in an inner loop.